cmd/tuf/app/sign.go

```diff
@@ -191,12 +189,35 @@ func SignMeta(ctx context.Context, store tuf.LocalStore, name string, signer sig
 		return err
 	}
 
-	sigs := make([]data.Signature, 0, len(s.Signatures))
+	// Get TUF public IDs associated to the signer.
+	var keyIDs []string
+	pubKey, err := keys.ConstructTufKey(ctx, signer, false)
+	if err != nil {
+		return err
+	}
+	keyIDs = append(keyIDs, pubKey.IDs()...)
+	if addDeprecatedKeyFormat {
+		oldPubKey, err := keys.ConstructTufKey(ctx, signer, true)
+		if err != nil {
+			return err
+		}
+		keyIDs = append(keyIDs, oldPubKey.IDs()...)
+	}
+
+	role := strings.TrimSuffix(name, ".json")
+	roleSigningKeys, err := repo.GetSigningKeyIDsForRole(role, store)
+	if err != nil {
+		return err
+	}
 
 	// Add it to your key entry
 	var added bool
-	for _, id := range key.IDs() {
-		// If pre-entries are defined.
+	sigs := make([]data.Signature, 0, len(s.Signatures))
+	for _, id := range keyIDs {
+		if _, ok := roleSigningKeys[id]; !ok {
```

Comment:
nit: Can you add a comment about why you added this check, for targets?

Reply:
Agree! Clear it was missing when I was chatting through it with you.